wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift

```diff
@@ -51,20 +52,22 @@ class ImagePickerManager: NSObject {
                                             message: nil,
                                             preferredStyle: .actionSheet)
 
-        // Choose from gallery
-        let galleryAction = UIAlertAction(title: Alert.choosePicture, style: .default) { [weak self] _ -> Void in
-            self?.sourceType = .photoLibrary
-            self?.getImage(fromSourceType: .photoLibrary)
-        }
-        actionSheet.addAction(galleryAction)
-
         // Take photo
         let cameraAction = UIAlertAction(title: Alert.takePicture, style: .default) { [weak self] _ -> Void in
             self?.sourceType = .camera
             self?.getImage(fromSourceType: .camera)
         }
         actionSheet.addAction(cameraAction)
 
+        guard !MediaShareRestrictionManager(sessionRestriction: ZMUserSession.shared()).isPhotoLibraryEnabled else {
+            let galleryAction = UIAlertAction(title: Alert.choosePicture, style: .default) { [weak self] _ -> Void in
+                self?.sourceType = .photoLibrary
+                self?.getImage(fromSourceType: .photoLibrary)
+            }
+            actionSheet.addAction(galleryAction)
+            return actionSheet
+        }
+
```

Comment:
1. Any particular reason for changing the order in the sheet?
2. You're missing the cancel button no?